test/src/test/java/com/rbmhtechnology/vind/test/ServerTest.java

```diff
@@ -2948,4 +2950,85 @@ public void testDeleteNonExistingDoc() {
 
         Assert.assertNotNull(result);
     }
+
+    @Test
+    @RunWithBackend({Solr, Elastic})
+    public void testSort() {
+
+        ZonedDateTime now = ZonedDateTime.now(ZoneId.of("UTC"));
+        ZonedDateTime yesterday = ZonedDateTime.now(ZoneId.of("UTC")).minus(1, ChronoUnit.DAYS);
+
+        FieldDescriptor<String> title = new FieldDescriptorBuilder()
+                .setFullText(true)
+                .setFacet(true)
+                .buildTextField("title");
+
+        SingleValueFieldDescriptor.DateFieldDescriptor<ZonedDateTime> created = new FieldDescriptorBuilder()
+                .setFacet(true)
+                .buildDateField("created");
+
+        SingleValueFieldDescriptor.UtilDateFieldDescriptor<Date> modified = new FieldDescriptorBuilder()
+                .setFacet(true)
+                .buildUtilDateField("modified");
+
+        NumericFieldDescriptor<Long> category = new FieldDescriptorBuilder()
+                .setFacet(true)
+                .buildMultivaluedNumericField("category", Long.class);
+
+        DocumentFactory assets = new DocumentFactoryBuilder("asset")
+                .addField(title)
+                .addField(created)
+                .addField(category)
+                .addField(modified)
+                .build();
+
+        Document d1 = assets.createDoc("1")
+                .setValue(title, "Hello World")
+                .setValue(created, yesterday)
+                .setValue(modified, new Date())
+                .setValues(category, Arrays.asList(1L, 2L));
+
+        Document d2 = assets.createDoc("2")
+                .setValue(title, "Hello Friends")
+                .setValue(created, now)
+                .setValue(modified, new Date())
+                .addValue(category, 4L);
+
+        SearchServer server = testBackend.getSearchServer();
+
+        server.index(d1);
+        server.index(d2);
+        server.commit();
+
+
+        FulltextSearch search = Search.fulltext("hello")
+                .sort(desc(score()));
+
+        SearchResult result = server.execute(search,assets);
+        assertEquals("1", result.getResults().get(0).getId());
+
+        search = Search.fulltext("hello")
+                .sort(asc(score()));
+
+        result = server.execute(search,assets);
+        assertEquals("1", result.getResults().get(0).getId());
+
+        search = Search.fulltext("hello")
+                .sort(asc(score()), desc(category));
+
+        result = server.execute(search,assets);
+        assertEquals("2", result.getResults().get(0).getId());
+
+        search = Search.fulltext("hello friends")
+                .sort(desc(score()));
+
+        result = server.execute(search,assets);
+        assertEquals("2", result.getResults().get(0).getId());
+
+        search = Search.fulltext("hello friends")
+                .sort(asc(score()));
+
+        result = server.execute(search,assets);
+        assertEquals("1", result.getResults().get(0).getId());
+    }
```

Comment:
maybe add a test for the combination of 2 sort params (like score and fieldsort)

Reply:
already in line 3017